chain/jsonrpc/src/errors.rs

```diff
@@ -36,12 +68,45 @@ impl RpcFrom<actix::MailboxError> for ServerError {
     }
 }
 
+impl RpcFrom<near_primitives::errors::InvalidTxError> for ServerError {
+    fn rpc_from(e: near_primitives::errors::InvalidTxError) -> ServerError {
+        ServerError::TxExecutionError(near_primitives::errors::TxExecutionError::InvalidTxError(e))
+    }
+}
+
+// -- block --
+
 impl RpcFrom<actix::MailboxError> for near_jsonrpc_primitives::types::blocks::RpcBlockError {
     fn rpc_from(error: actix::MailboxError) -> Self {
         Self::InternalError { error_message: error.to_string() }
     }
 }
 
+impl RpcFrom<near_client_primitives::types::GetBlockError>
+    for near_jsonrpc_primitives::types::blocks::RpcBlockError
+{
+    fn rpc_from(error: near_client_primitives::types::GetBlockError) -> Self {
+        match error {
+            near_client_primitives::types::GetBlockError::UnknownBlock { error_message } => {
+                Self::UnknownBlock { error_message }
+            }
+            near_client_primitives::types::GetBlockError::NotSyncedYet => Self::NotSyncedYet,
+            near_client_primitives::types::GetBlockError::IOError { error_message } => {
+                Self::InternalError { error_message }
+            }
+            near_client_primitives::types::GetBlockError::Unreachable { ref error_message } => {
+                tracing::warn!(target: "jsonrpc", "Unreachable error occurred: {}", &error_message);
+                crate::metrics::RPC_UNREACHABLE_ERROR_COUNT
+                    .with_label_values(&["RpcBlockError"])
+                    .inc();
+                Self::InternalError { error_message: error.to_string() }
+            }
+        }
+    }
+}
+
+// -- block / changes --
```

Comment:
Whenever I see comments to segment file, I question myself whether it makes anyone's life easier compared to having those blocks in different files. How do you feel about having `errors` folder with this file split into several files?

Reply:
I thought the same, but looking at it, I think since it's all about implementation, it's fine to have this as-is. Since there are no special structures or anything.

Reply:
EDIT: I'm doing this next. I'll merge the `parser` and `errors`.